asf_public_discourse_home_decarbonisation/pipeline/data_processing_flows/processing_text_data.py

```diff
@@ -0,0 +1,163 @@
+"""
+A Flow for processing title and text data from MSE sub-forums.
+python asf_public_discourse_home_decarbonisation/pipeline/data_processing_flows/processing_text_data.py --datastore=s3 --package-suffixes=.txt run --max-num-splits 2000 --max-workers 100
+"""
+
+import os
+
+# Upgrading pip and installing requirements
+os.system("python -m pip install --upgrade pip")
+os.system(
+    f"pip install -r {os.path.dirname(os.path.realpath(__file__))}/flow_requirements.txt 1> /dev/null"
+)
+os.system("python -m spacy download en_core_web_sm")
+
+from metaflow import FlowSpec, step, batch, Parameter
+
+S3_BUCKET = "asf-public-discourse-home-decarbonisation"
+MSE_S3_PROCESSED_OUTPUTS_FOLDER_PATH = "data/mse/outputs/processed"
+CHUNKSIZE = 10000
+
+
+class TextProcessingFlow(FlowSpec):
+    category = Parameter(
+        name="mse_category",
+        help="An MSE category or sub-forum",
+        default="green-ethical-moneysaving",
+    )
+
+    batch_date = Parameter(
+        name="batch",
+        help='Batch name (e.g. datetime of collection "2023_11_15" or "newest" for the most up to date)',
+        default="2023_11_15",
+    )
+
+    test = Parameter(
+        name="test",
+        help="Set to True to run flow on test mode.",
+        default=False,
+    )
+
+    @step
+    def start(self):
+        """
+        Starts the flow.
+        """
+        from flow_utils import (
+            get_mse_data,
+        )
+
+        self.mse_data = get_mse_data(self.category, self.batch_date)
+
+        self.next(self.process_text_and_titles)
+
+    @step
+    def process_text_and_titles(self):
+        """
+        Pre-processing text and titles before applying lemmatisation by
+        removing URLs, putting text to lowercase and removing username patterns.
+        """
+        from flow_utils import process_text
+
+        self.mse_data["processed_text"] = self.mse_data["text"].apply(
+            lambda x: process_text(x)
+        )
+        self.mse_data["processed_title"] = self.mse_data["title"].apply(
+            lambda x: process_text(x)
+        )
+
+        self.next(self.prepare_for_lemmatising)
+
+    @batch
```

Comment:
Is it necessary to run the "prepare_for_lemmatising" step on batch?

Reply:
just avoids my computer being really slow